tcs/core/tcs_trusted_worker_manager/enclave/work_order_processor.cpp

```diff
@@ -213,31 +213,28 @@ namespace tcf {
 
         if (data_items_in.size() > 0) {
             // data_items_in vector is sorted based out of index
-            // data at index 0 is the identifier of the workorder
-            WorkOrderDataHandler& code_data_item = data_items_in.at(0);
-            // Vector is sorted the first object should have index 0
-            if (code_data_item.workorder_data.index == 0) {
-                for (auto d : data_items_in) {
-                    in_wo_data.emplace_back(d.workorder_data.index, d.workorder_data.decrypted_data);
-                }
-
-                for (auto d : data_items_out) {
-                    out_wo_data.emplace_back(d.workorder_data.index, d.workorder_data.decrypted_data);
-                }
-
-                WorkloadProcessor processor;
-                std::string workload_type = ByteArrayToStr(code_data_item.workorder_data.decrypted_data);
-                processor.ProcessWorkOrder(
+            for (auto d : data_items_in) {
+                in_wo_data.emplace_back(d.workorder_data.index, d.workorder_data.decrypted_data);
+            }
+
+            for (auto d : data_items_out) {
+                out_wo_data.emplace_back(d.workorder_data.index, d.workorder_data.decrypted_data);
+            }
+
+            WorkloadProcessor processor;
+            std::string workload_str;
+            HexStringToBinary((uint8_t *)workload_str.c_str(), work_load_id.length(), work_load_id);
```

Comment:
Where is memory allocation done for first argument of HexStringToBinary

Reply:
I will correct it.